pkg/crinit/crinit.go

```diff
@@ -28,21 +29,15 @@ import (
 
 // NewClusterregistryCommand creates the `clusterregistry` command.
 func NewClusterregistryCommand(out io.Writer, defaultServerImage, defaultEtcdImage string) *cobra.Command {
-	cmds := &cobra.Command{
+	rootCmd := &cobra.Command{
 		Use:   "crinit",
 		Short: "crinit runs a cluster registry in a Kubernetes cluster",
 		Long:  "crinit bootstraps and runs a cluster registry as a Deployment in an existing Kubernetes cluster.",
-		Run:   runHelp,
 	}
 
-	// From this point and forward we get warnings on flags that contain "_" separators
-	cmds.SetGlobalNormalizationFunc(flag.WarnWordSepNormalizeFunc)
```

Comment:
This line is the reason we issue a warning about logging before parsing. It's because we have set the warn function as the normalization function when we attempt to convert flags to use dashes instead of underscores before we've parsed our flags.

Reply:
Ah, that makes sense. In any case, it seems like the tweak in `cmd/crinit/crinit.go` is the canonical workaround.